packages/resource-deployment/scripts/install-parallel.sh

```diff
@@ -0,0 +1,145 @@
+# Licensed under the MIT License.
+
+# shellcheck disable=SC1090
+set -eo pipefail
+
+export appInsightsKey
+export apiManagementName
+export batchAccountName
+export cosmosAccountName
+export datalakeStorageAccountName
+export dropFolder="${0%/*}/../../../"
+export environment
+export functionAppName
+export keyVault
+export keyVaultUrl
+export location
+export resourceGroupName
+export subscription
+export storageAccountName
+export clientId
+export templatesFolder="${0%/*}/../templates/"
+export apiTemplates="$templatesFolder"rest-api-templates
+
+exitWithUsageInfo() {
+    echo "
+Usage: $0 -e <environment> -l <Azure region> -o <organisation name> -p <publisher email> -r <resource group> -s <subscription name or id> -c <client id>
+where:
+Resource group - The name of the resource group that everything will be deployed in.
+Subscription - The subscription for the resource group.
+Environment - The environment in which the set up is running.
+Publisher email - The email for notifications.
+Resource group - The resource group that this API instance needs to be added to.
+Client ID - the app registration client id used for function app authentication/ authorization.
+Azure region - Azure region where the instances will be deployed. Available Azure regions:
+    centralus
+    eastasia
+    southeastasia
+    eastus
+    eastus2
+    westus
+    westus2
+    northcentralus
+    southcentralus
+    westcentralus
+    northeurope
+    westeurope
+    japaneast
+    japanwest
+    brazilsouth
+    australiasoutheast
+    australiaeast
+    westindia
+    southindia
+    centralindia
+    canadacentral
+    canadaeast
+    uksouth
+    ukwest
+    koreacentral
+    koreasouth
+    francecentral
+    southafricanorth
+    uaenorth
+"
+    exit 1
+}
+
+
+function waitForProcesses(){
+    processesToWaitFor=$1
+    for pid in ${processesToWaitFor[@]}; do
+        wait $pid
+        returnCode=$?
+        echo "Value of pid is $pid, Return code is $returnCode"
+    done
+}
+
+# Read script arguments
+while getopts ":r:s:l:e:o:p:c:" option; do
+    case $option in
+    r) resourceGroupName=${OPTARG} ;;
+    s) subscription=${OPTARG} ;;
+    l) location=${OPTARG} ;;
+    e) environment=${OPTARG} ;;
+    o) orgName=${OPTARG} ;;
+    p) publisherEmail=${OPTARG} ;;
+    c) clientId=${OPTARG} ;;
+    *) exitWithUsageInfo ;;
+    esac
+done
+
+if [[ -z $resourceGroupName ]] || [[ -z $subscription ]] || [[ -z $environment ]] || [[ -z $orgName ]] || [[ -z $publisherEmail ]]; then
+    exitWithUsageInfo
+fi
+
+# Login to Azure if required
+if ! az account show 1>/dev/null; then
+    az login
+fi
+
+az account set --subscription "$subscription"
+
+. "${0%/*}/create-resource-group.sh"
+
+. "${0%/*}/create-storage-account.sh"
+
+echo "Finished creating storage account"
+
+resourceGroupSuffix=${storageAccountName:11}
+cosmosAccountName="allycosmos$resourceGroupSuffix"
+apiManagementName="apim-a11y$resourceGroupSuffix"
+
+. "${0%/*}/create-api-management.sh" &
```

Comment:
it may not be a good idea to source the script when you are parallelizing them. it may have unpredictable state on the parent process' environment variables.

Reply:
if we are planning to parallelize, we may need to refactor out the code that sets environment variables out & not parallelize that part.

Reply:
background process cannot touch the parents variable at all !